netbox_branching/models/branches.py

```diff
@@ -501,6 +512,730 @@ def migration_progress_callback(action, migration=None, fake=False):
 
     migrate.alters_data = True
 
+    # Helper class and functions for collapsing ObjectChanges during merge
+    class CollapsedChange:
+        """
+        Represents a collapsed set of ObjectChanges for a single object.
+        """
+        def __init__(self, key, model_class):
+            self.key = key  # (content_type_id, object_id)
+            self.model_class = model_class
+            self.changes = []  # List of ObjectChange instances, ordered by time
+            self.final_action = None  # 'create', 'update', 'delete', or 'skip'
+            self.prechange_data = None
+            self.postchange_data = None
+            self.last_change = None  # The most recent ObjectChange (for metadata)
+
+            # Dependencies for ordering
+            self.depends_on = set()  # Set of keys this change depends on
+            self.depended_by = set()  # Set of keys that depend on this change
+
+        def __repr__(self):
+            obj_id = self.key[1]
+            suffix = f" ({self.key[2]})" if len(self.key) > 2 else ""
+            return (
+                f"<CollapsedChange {self.model_class.__name__}:{obj_id}{suffix} "
+                f"action={self.final_action} changes={len(self.changes)}>"
+            )
+
+    @staticmethod
+    def _diff_object_change_data(action, prechange_data, postchange_data):
```

Comment:
Can we convert all these static methods to standalone utility functions? They don't seem particularly relevant to the branch itself.